### ISSUE TITLE
Edge Pinned tabs autolocking when they should not be.

### ISSUE THREAD

**Issue by Riversiderepeat** [gabrielmaldi/chrome-lock-tab#10]
Just what it says. In options I have "Autolock pinned tabs unchecked," yet pinned tabs keep getting locked anyway. Very annoying.

**Comment by gabrielmaldi** [comment]
Hello, @Riversiderepeat. I just tested the Auto-lock pinned tabs feature, both in Chrome and Edge, and it seems to be working fine for me. Have you tried closing the browser and reopening it? Can you share a screen recording which shows how to reproduce the issue with a single pinned tab? Thanks!

**Comment by Riversiderepeat** [comment]
Are you using tabs for shopping (amazon, local grocery stores etc.? And actually trying to place things in carts, or search? How exactly are you testing it?

**Comment by gabrielmaldi** [comment]
Test scenario 1:

1. Enable "Auto-lock pinned tabs" option in the extension's setting
2. Pin a tab and interact with it (i.e. click somewhere on the page)
3. Try to close the tab or navigate to another page (same or different domain, doesn't matter)
4. Verify that the tab displays a warning modal which prevents the tab from being closed or navigated away ✅

Test scenario 2:

1. Disable the setting
2. Same
2. Same
4. Verify that the tab closes/navigates without warning ✅

**Comment by gabrielmaldi** [comment]
I think that what you're experiencing is not related to pinned tabs. We use the onbeforeunload event which doesn't distinguish between closing a tab, navigating to another website, navigating to another page in the same website, posting a form, etc. So if the tab you're locking, instead of making XHR/fetch requests to update the shopping cart, posts a form or navigates to another subpage, onbeforeunload will kick in and you'll see the warning. See: https://github.com/gabrielmaldi/chrome-lock-tab/issues/6#issuecomment-1863257263

**Comment by Riversiderepeat** [comment]
Yes, I have restarted Edge. Now I can't get a pinned tab to lock at all so I'm unable to provide screen shots until get that working again.

**Comment by Riversiderepeat** [comment]
Okay so here's a shot of the settings showing the autolock option turned off, and a quick video of one pinned tab, already locked, being unlocked, then a search performed on that tab. Then don't blink because for a split second at the end, you can see the tab locked again after the search.

Also, oddly I noticed during this process that one tab actually displayed a double lock icon, but that cleared up after a browser restart.
![double lock](https://github.com/user-attachments/assets/4942f51e-cbfc-450b-8b9a-47d84ef3c4fd)
![settings](https://github.com/user-attachments/assets/f868c483-ba9f-4e37-a865-103267734165)

https://github.com/user-attachments/assets/7dcb2a4f-0a49-44c7-8c8c-9f1b26c0de7e

**Comment by Riversiderepeat** [comment]
Accidentally closed this.

**Comment by gabrielmaldi** [comment]
Thanks for the info.

I tried reproducing but wasn't able to.

Is it possible that you didn't hit the Save button after unchecking Auto-lock pinned tabs? I'm asking because, from what I'm seeing in your video and my testing of hannaford.com, the site does a full document navigation when you search; and that suggests that the extensions is locking the tab again by checking its settings (which could be either the auto-lock pinned tabs option, or rules-based matching by URL which I see you only set up for Gmail.)

To test on hannaford.com I had to use Opera instead of Edge because the site blocks access from where I am, and Opera's free VPN allows me to access the site while Edge's doesn't. But I did test other similar sites on Edge (and also couldn't reproduce.)

Try the following and let me know how it goes:

1. Uncheck Auto-lock pinned tabs and hit Save
2. Fully close Edge (all its windows and tabs)
3. Open Edge again and verify that Auto-lock pinned tabs remains unchecked (important)
4. Go to the pinned tab where you have hannaford.com
5. Perform a search
6. Does the tab lock itself again?
7. If it does (because you see the 🔒 icon or the Lock Tab's icon switches to locked), do you get the `Leave site?` prompt (remember to click somewhere on the page before trying to navigate away)?

Thanks

**Comment by gabrielmaldi** [comment]
Closing due to inactivity.
